cmake/SetupCompilers.cmake

```diff
@@ -53,6 +53,10 @@ elseif (CMAKE_CXX_COMPILER_ID MATCHES GNU)
 elseif (CMAKE_CXX_COMPILER_ID MATCHES Intel)
   set(RAJA_COMPILER "RAJA_COMPILER_ICC")
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+elseif(CMAKE_CXX_COMPILER_ID MATCHES XL)
+  set(RAJA_COMPILER "RAJA_COMPILER_XLC12")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O5")
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
This should be set in cache.  I realize the intel compiler version doesn't, but they both should.  Just add `CACHE STRING "C++ compiler flags"` at the end of the set arguments.